### PR TITLE
AVRO-1858 add tojson head mode

### PR DESCRIPTION
**Pull request by MikeHurleySurescripts**
Let me know if this pull request should be pointed to a different branch.

### CODE REVIEW

File: lang/java/tools/src/main/java/org/apache/avro/tool/DataFileReadTool.java [hunk 1]
```diff
@@ -84,8 +94,27 @@ public int run(InputStream stdin, PrintStream out, PrintStream err,
...
 
+  private static long getHeadCount(OptionSet optionSet, OptionSpec<String> headOption, List<String> nargs) {
+    long headCount = Long.MAX_VALUE;
+    if(optionSet.has(headOption)) {
+      headCount = DEFAULT_HEAD_COUNT;
+      List<String> headValues = optionSet.valuesOf(headOption);
+      if(headValues.size() > 0) {
+        // if the value parses to int, assume it's meant to go with --head
+        // otherwise assume it was an optionSet.nonOptionArgument and add back to the list
+        // TODO: support input filenames whose whole path+name is int parsable?
```
Comment (busbey) [inline lang/java/tools/src/main/java/org/apache/avro/tool/DataFileReadTool.java]: isn't there a flag that can separate optional args from the nonOptionArgument? something like `--head -- 1`

File: lang/java/tools/src/test/java/org/apache/avro/tool/TestDataFileTools.java [hunk 2]
```diff
@@ -118,6 +118,33 @@ public void testReadToJsonPretty() throws Exception {
...
+    assertEquals(expectedJson,
+      run(new DataFileReadTool(), "--head", "5", sampleFile.getPath()));
+  }
+
+  @Test
+  public void testReadHeadLongCount() throws Exception {
+    assertEquals(jsonData,
+      run(new DataFileReadTool(), "--head=3000000000", sampleFile.getPath()));
+  }
+
```
Comment (busbey) [inline lang/java/tools/src/test/java/org/apache/avro/tool/TestDataFileTools.java]: could you add a test for a negative number and for 0?

File: lang/java/tools/src/main/java/org/apache/avro/tool/DataFileReadTool.java (hunk 1 above)
Comment (MikeHurleySurescripts) [inline lang/java/tools/src/main/java/org/apache/avro/tool/DataFileReadTool.java]: I've never seen anything like that. Seems odd. I'd prefer forcing the --head=X syntax but I couldn't find a way to do that with the command line arg library this tool uses.

File: lang/java/tools/src/test/java/org/apache/avro/tool/TestDataFileTools.java (hunk 2 above)
Comment (MikeHurleySurescripts) [inline lang/java/tools/src/test/java/org/apache/avro/tool/TestDataFileTools.java]: What would people prefer happens? Should zero and negative simply do nothing? Should they throw an exception saying "bad input"?

Comment (busbey) [inline lang/java/tools/src/test/java/org/apache/avro/tool/TestDataFileTools.java]: with 0, I'd expect just no lines of output and success.

with negative, we can either go with posix ([it's an error](http://www.unix.com/man-page/posix/1p/head/)) or the gnu coreutils route and have negative mean "all but the last _n_ lines".

Personally, I'd favor aping posix here.
